sirepo/package_data/static/js/sirepo.js

```diff
@@ -2161,6 +2166,31 @@ SIREPO.app.factory('requestSender', function(cookieService, errorService, userAg
         $window.open(self.formatUrl(routeName, params), '_blank');
     };
 
+    self.openSimulation = (app, route, simId) => {
```

Comment:
All this information is in the GUI in the route map. I don't think this should (or does) change from app to app.  You definitely don't need a new API.

Why do you need the schema at this point?

Reply:
We want the "data" route, which does not exist in srw's localMap. I was previously assuming the form of the route (`/${route}/:simulationId`). Maybe that's actually a safe assumption since addRoute references it:
https://github.com/radiasoft/sirepo/blob/fe4a0b7512432acd62db91d6ee02deec8ff4b67d/sirepo/package_data/static/js/sirepo.js#L204-L207
but using the schema seems to be "the right way". One way or another the route has to get added to localMap.

Reply:
Yes, I see the problem now. I think what we want is a generic local route for this so we avoid stuff like this:

$scope.route || 'visualization',

We could have a route alias that is know globally. However, that's even more complicated.

Perhaps this does need to be a new entrypoint, but with all the logic server side, generating the local route and redirecting at the end, e.g.

window.open(self.formatUrl("simulationRedirect", {code: "srw", simId: "xyz"}, '_blank');

Thoughts?

Reply:
Sounds good; I'll flesh that out